cli/admin/src/test/java/io/pravega/cli/admin/AbstractTlsAdminCommandTest.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.cli.admin;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.stream.impl.DefaultCredentials;
+import io.pravega.local.LocalPravegaEmulator;
+import io.pravega.test.common.SecurityConfigDefaults;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.Timeout;
+
+import java.net.URI;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+public abstract class AbstractTlsAdminCommandTest {
+
+    // Security related flags and instantiate local pravega server.
+    protected static final AtomicReference<AdminCommandState> STATE = new AtomicReference<>();
+
+    private static final Integer CONTROLLER_PORT = 9090;
+    private static final Integer SEGMENT_STORE_PORT = 6000;
+    private static final Integer REST_SERVER_PORT = 9091;
+
+    @Rule
+    public final Timeout globalTimeout = new Timeout(80, TimeUnit.SECONDS);
+
+    protected boolean authEnabled = false;
```

Comment:
Any reason why these shouldn't be `static` or `final`?

Reply:
`@Rule` requires the variable to be `final` only and not `static`. The variables `authEnabled` and `tlsEnabled` cannot be `final` as these flags are set before calling `super.setUp()` in the test classes based on whether we test with authentication or TLS enabled or both.
Also static non final variables cannot be declared according to the checkstyle.